lib/matplotlib/gridspec.py

```diff
@@ -182,7 +195,7 @@ class GridSpec(GridSpecBase):
     as the SubplotParams.
     """
 
-    def __init__(self, nrows, ncols,
+    def __init__(self, nrows, ncols, fig=None,
```

Comment:
This technically is backwards incompatible (although I'm totally fine with the change), so should be noted specifically in the API changes (perhaps it already is, haven't checked).

Reply:
This is definitely still a todo.  I was assuming this would take some time to get into reasonable shape before dotting the i-s.

Reply:
No worries, I just skimmed through the whole thing and left notes here and there.  If we want to keep strict backcompatibility here (not saying that we necessarily should), the trick at https://github.com/matplotlib/matplotlib/pull/9110/files#diff-7e79bc5b4cdd21de353697e9ada248b7R2008 could be useful.